assets/src/edit-story/components/inspector/prepublish/prepublishChecklistProvider.js

```diff
@@ -61,9 +66,28 @@ function PrepublishChecklistProvider({ children }) {
     }
   }, [handleRefreshList, refreshOnInitialLoad, refreshOnPageSizeChange]);
 
+  const [checkpointState, dispatch] = useReducer(
+    checkpointReducer,
+    PPC_CHECKPOINT_STATE.ONLY_RECOMMENDED
+  );
+
+  // Check for different qualifications to be met to update current PPC checkpoint
+  // 1. Story is no longer empty (ON_DIRTY_STORY)
+  // 2. Publish button is hit on a draft (ON_PUBLISH_CLICKED)
+  // 3. Story has more than 4 pages
+  useEffect(() => {
+    if (story.pages.length > 4) {
+      dispatch(PPC_CHECKPOINT_ACTION.ON_STORY_HAS_5_PAGES);
+    }
+  }, [story?.pages]);
```

Comment:
Nice! with the other PR I'll put up in the next hr or so we can just make this an event listener and not need the conditional I think. (would basically do this conditional check in the code responsible for looking at story and dispatching event)
```
    useStoryTriggerListener(
      STORY_EVENTS.onEvaluationReady, // This can be called whatever
      useCallback(() => {
        dispatch(PPC_CHECKPOINT_ACTION.ON_STORY_HAS_5_PAGES);
      }, [])
    )
```

Reply:
_beautiful_